overlord/ifacestate/handlers.go

```diff
@@ -359,23 +366,36 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	var plugSnapst snapstate.SnapState
-	if err := snapstate.Get(st, connRef.PlugRef.Snap, &plugSnapst); err != nil {
-		return err
-	}
-
 	var slotSnapst snapstate.SnapState
-	if err := snapstate.Get(st, connRef.SlotRef.Snap, &slotSnapst); err != nil {
+	err = snapstate.Get(st, connRef.SlotRef.Snap, &slotSnapst)
+	if err != nil && err != state.ErrNoState {
 		return err
```

Comment:
I don't think we want to skip this. That's an action on doConnect, which can only be initiated by an explicit action, right? In those cases we really want to error out, because the precise action that was requested is failing and won't work as expected.

If so, instead of changing that now, let's simply do this for now:

`return fmt.Errorf("internal error: cannot get state of snap %q: %v", connRef.SlotRef.Snap, err)`

Rest of the logic can remain unchanged (no need to check for ErrNoState right above, since it's an internal error and we're adding context).

Same below in the other two cases.